src/components/sidebar/ContextAwareSidebar.tsx

```diff
@@ -360,24 +353,44 @@ class ContextAwareSidebar
                     <div className="container">
 
                       {
-                        refs.map(ref => (
-                          <div key={ref.guid} className="ref-thing">
+                        refs.map(ref => getRefResourceFromRef(ref))
+                        .filter(res => res.resourceState !== ResourceState.DELETED)
```

Comment:
This will throw an error if `res` is undefined because it wasn't present in `resourcesById`

Reply:
filtered out undefined res